### PR TITLE
Firestore Cloud implementation + Native Favicon and Splash Screen

### PR DESCRIPTION
**Pull request by marcosvonstremayr**
During this module I implemented Firestore Cloud into the app. Now when you fetch data from the API, it stores it into the database and cache so it's available without internet connection. When there isn't data stored in the cache or database, the app shows an empty screen. Also I added the app icon and a splash screen, 

ezgif-4-cc6ffbdd1a

image

### CODE REVIEW

File: lib/src/core/util/assets_constants.dart [hunk 1]
```diff
@@ -4,4 +4,5 @@ abstract class AssetsConstants {
   static const String pathToPoster = "assets/images/poster.webp";
   static const String backCardImage = "assets/images/back_card.webp";
   static const String errorScreen = "assets/images/error_screen.webp";
+  static const String emptyScreenImg = "assets/images/empty.webp";
```
Comment (MarcosCGlobant) [inline lib/src/core/util/assets_constants.dart]: you can use the complete word, like the constant above

File: lib/src/core/util/dimensions_constants.dart [hunk 2]
```diff
@@ -31,4 +31,7 @@ abstract class DimensionsConstants {
   static const double cardDetailImgPadding = 15;
   static const double drawerSubcategoriesFontSize = 18;
   static const double errorMsgFontSize = 24;
+  static const double emptyCardsFontSize = 24;
+  static const double emptyImgHeight = 500;
```
Comment (MarcosCGlobant) [inline lib/src/core/util/dimensions_constants.dart]: use the full word for the constant, please

File: lib/src/core/util/assets_constants.dart (hunk 1 above)
Comment (marcosvonstremayr) [reply lib/src/core/util/assets_constants.dart]: Done

File: lib/src/core/util/dimensions_constants.dart (hunk 2 above)
Comment (marcosvonstremayr) [reply lib/src/core/util/dimensions_constants.dart]: Done

**Review by MarcosCGlobant — APPROVED** [review]

File: lib/main.dart [hunk 3]
```diff
@@ -49,16 +52,21 @@ class _HearthstoneAppState extends State<HearthstoneApp> {
...
-        fontFamily: StringConstants.appFont,
-        scaffoldBackgroundColor: Constants.appBackgroundColor,
-        canvasColor: Constants.appCanvasColor,
-      ),
-      home: Homepage(
-        bloc: _bloc,
-      ),
+    return FutureBuilder(
+      future: Firebase.initializeApp(),
+      builder: (BuildContext context, AsyncSnapshot<dynamic> snapshot) {
```
Comment (FranciscoGaraventa) [inline lib/main.dart]: trailing comma

**Review by FranciscoGaraventa — APPROVED** [review]
LG, approve with minor comment

File: lib/main.dart (hunk 3 above)
Comment (marcosvonstremayr) [reply lib/main.dart]: done

**Pull request merged**